gov/politeia/proposals.go

```diff
@@ -90,371 +87,509 @@ func NewProposalsDB(politeiaURL, dbPath string) (*ProposalDB, error) {
+		}
+		proposal.UserID = um.UserID
+
+		// Comments count
+		proposal.CommentsCount =
+			int32(commentsCounts[record.CensorshipRecord.Token])
+
+		// Vote summary data
+		summary := voteSummaries[proposal.Token]
+		proposal.VoteStatus = summary.Status
+		proposal.VoteResults = summary.Results
+		proposal.EligibleTickets = summary.EligibleTickets
+		proposal.StartBlockHeight = summary.StartBlockHeight
+		proposal.EndBlockHeight = summary.EndBlockHeight
+		proposal.QuorumPercentage = summary.QuorumPercentage
+		proposal.PassPercentage = summary.PassPercentage
+
+		var totalVotes uint64
+		for _, v := range summary.Results {
+			totalVotes += v.Votes
+		}
+		proposal.TotalVotes = totalVotes
+
+		// Status change metadata
+		ts, changeMsg, err := statusChangeMetadataDecode(record.Metadata)
+		if err != nil {
+			return nil, fmt.Errorf("statusChangeMetadataDecode err: %v", err)
+		}
+		proposal.PublishedAt = ts[0]
+		proposal.CensoredAt = ts[1]
+		proposal.AbandonedAt = ts[2]
+		proposal.StatusChangeMsg = changeMsg
+
+		// Append proposal after inserting the relevant data
+		proposals = append(proposals, &proposal)
 	}
 
-	defer atomic.StoreInt64(&db.lastSync, time.Now().UTC().Unix())
+	return proposals, nil
+}
 
-	// Retrieve and update all current proposals whose vote statuses is either
-	// NotAuthorized, Authorized and Started
-	numRecords, err := db.updateInProgressProposals()
+func (db *ProposalsDB) fetchTicketVoteResults(token string) (*pitypes.ProposalChartData, error) {
+	// Fetch ticket votes details to acquire vote bits options info.
+	details, err := db.client.TicketVoteDetails(ticketvotev1.Details{
+		Token: token,
+	})
 	if err != nil {
-		return err
+		return nil, fmt.Errorf("Pi client TicketVoteDetails err: %v", err)
 	}
 
-	// Retrieve and update any new proposals created since the previous
-	// proposals were stored in the db.
-	lastProposal, err := db.lastSavedProposal()
-	if err != nil && err != storm.ErrNotFound {
-		return fmt.Errorf("lastSavedProposal failed: %v", err)
+	// Maps the vote bits option to their respective string ID.
+	voteOptsMap := make(map[uint64]string)
+	for _, opt := range details.Vote.Params.Options {
+		voteOptsMap[opt.Bit] = opt.ID
 	}
 
-	var queryParam string
-	if len(lastProposal) > 0 && lastProposal[0].TokenVal != "" {
-		queryParam = fmt.Sprintf("?before=%s", lastProposal[0].TokenVal)
-	}
-	publicProposals, err := db.fetchAPIData(queryParam)
+	tvr, err := db.client.TicketVoteResults(ticketvotev1.Results{
+		Token: token,
+	})
 	if err != nil {
-		return err
+		return nil, fmt.Errorf("Pi client TicketVoteResults err: %v", err)
 	}
 
-	n, err := db.saveProposals(publicProposals)
-	if err != nil {
-		return err
+	// Parse reply to proposal chart data
+	var chart pitypes.ProposalChartData
+	var timestamps []int64
+	for _, v := range tvr.Votes {
+		bit, err := strconv.ParseUint(v.VoteBit, 16, 64)
+		if err != nil {
+			return nil, err
+		}
+		switch voteOptsMap[bit] {
```

Comment:
I think we should be doing a bitwise `&` with the bit options otherwise extraneous bits would break the switch.